docs/design/proposals/server-side-override/server-side-override.md

```diff
@@ -0,0 +1,224 @@
+    
+    ```yaml
+    services.k8s.aws/spec-overrides: 
+       “{ 
+           \“overrideFieldB\”:
+               {
+                   \“userInput\”:\“valueB\”,
+                   \“serverOverride\”:\“valueB-xyz\”
+               }
+       }”
+       Spec:
+        fieldA: valueA
+        overrideFieldB: valueB
+       Status:
+        fieldC: valueC
+        conditions:
+        - type: ACK.ServerSideOverride
+          status: True
+          reason: ServerSideSpecOverride
+          message: Spec override values are present in services.aws.k8s/spec-overrides annotation
+    ```
+   
+    > Note: the annotation above has single string value. It is been broken into multiple lines to showcase the structure better.
+
+3. During the next reconciliation of manifest mentioned in #1, the desired state will be constructed not just from spec 
+   in etcd but also from the `services.k8s.aws/spec-overrides` Annotations. The desired object will look like the object
+   persisted in etcd in #2
+
+   After reading the desired state from etcd, the reconciler code will go through all the spec fields and see whether 
+   they exist in the spec-override annotation map. If a corresponding entry exists, reconciler will marshall current 
+   desired value of spec field and compare with “userInput” in annotation override, 
+
+   * If there is no difference between “userInput” and marshalled desiredValue, reconciler will update the desired state
+     with serverOverride from spec-override annotation. 
+
+   * If the marshalled string does `not` match “userInput”, it means there was some change in manifest by the user, and 
+     serverOverrides from the spec-override map will not be used for desired state.
+     
+   In this case, There will be no delta between desired and latest object, And no AWS updates will be triggered.
+
+4. Now let’s assume that User updates the manifest. The manifest will look like
+
+    ```yaml
+    Spec:
+     fieldA: valueA
+     overrideFieldB: updatedValueB
+    ```
+
+    Now, during reconciliation, when the desired object is being constructed by parsing Annotations, reconciler will see
+    that overrideFieldB was updated because the userInput inside AnnotationMap is not equal to the current value in manifest.
+    Reconciler will remove the annotation for overrideFieldB.
+    
+5. At this time, the desired object looks different from latest object in AWS and an Update will happen. The result of 
+   UpdateOutput will contain following fields, fieldA → “valueA”, overrideFieldB → “updatedValueB-xyz”
+
+   During sdkUpdate, Annotations will be updated again. 
+
+   The object persisted in etcd will look like
+
+    ```yaml
+    Annotations:
+    services.k8s.aws/spec-overrides: 
+    “{
+        \“overrideFieldB\”:
+            {
+                \“userInput\”:\“updatedValueB\”,
+                \“serverOverride\”:\“updatedValueB-xyz\”
+            }
+    }”
+    Spec:
+     fieldA: valueA
+     overrideFieldB: updatedValueB-xyz
```

Comment:
This should be `updatedValueB`, right? We don't patch over the user's Spec input?